wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsdfu/DFUFileAccessInfoWrapper.java

```diff
@@ -140,11 +142,17 @@ public DFUFileAccessInfoWrapper(org.hpccsystems.ws.client.gen.axis2.wsdfu.v1_39.
             availableLocations.put(fileLocations[locationindex].getLocationIndex(), fileLocations[locationindex].getHost());
         }
 
-        DFUFilePartWrapper [] wrappedDFUFileParts = new DFUFilePartWrapper[fileParts.length];
-        for (int i = 0; i < fileParts.length; i++)
+        int dataFilePartsCount = fileParts.length;
+        if (fileType == DFUFileTypeWrapper.Index && dataFilePartsCount % availableLocations.size() != 0) // apparently not all keyed files report TLK as file part
```

Comment:
I agree with you on this it does seem a little hacky, but I think it should always work.

Reply:
yep, now that I think about it, I need to at least make sure we log this